pennylane/interfaces/jax_jit_tuple.py

```diff
@@ -0,0 +1,297 @@
+            tapes=tapes,
+            device=device,
+            execute_fn=execute_fn,
+            gradient_kwargs=gradient_kwargs,
+            _n=_n,
+        )
+
+    return _execute_bwd_tuple(
+        parameters,
+        tapes=tapes,
+        device=device,
+        execute_fn=execute_fn,
+        gradient_fn=gradient_fn,
+        gradient_kwargs=gradient_kwargs,
+        _n=_n,
+    )
+
+
+def _execute_bwd_tuple(
+    params,
+    tapes=None,
+    device=None,
+    execute_fn=None,
+    gradient_fn=None,
+    gradient_kwargs=None,
+    _n=1,
+):  # pylint: disable=dangerous-default-value,unused-argument
+
+    # Copy a given tape with operations and set parameters
+    def _copy_tape(t, a):
+        tc = t.copy(copy_operations=True)
+        tc.set_parameters(a)
+        return tc
+
+    @jax.custom_jvp
+    def execute_wrapper(params):
+        def wrapper(p):
+            """Compute the forward pass."""
+            new_tapes = [_copy_tape(t, a) for t, a in zip(tapes, p)]
+            with qml.tape.Unwrap(*new_tapes):
+                res, _ = execute_fn(new_tapes, **gradient_kwargs)
+            return res
+
+        shape_dtype_structs = _tapes_shape_dtype_tuple(tapes, device)
+        res = jax.pure_callback(wrapper, shape_dtype_structs, params)
+        return res
+
+    def _grad_transform_jac_via_callback(params, device):
+        """Perform a callback to compute the jacobian of tapes using a gradient transform (e.g., parameter-shift or
+        finite differences grad transform).
+
+        Note: we are not using the batch_jvp pipeline and rather split the steps of unwrapping tapes and the JVP
+        computation because:
+
+        1. Tape unwrapping has to happen in the callback: otherwise jitting is broken and Tracer objects
+        are converted to NumPy, something that raises an error;
+
+        2. Passing in the tangents as an argument to the wrapper function called by the jax.pure_callback raises an
+        error (as of jax and jaxlib 0.3.25):
+        ValueError: Pure callbacks do not support transpose. Please use jax.custom_vjp to use callbacks while
+        taking gradients.
+
+        Solution: Use the callback to compute the jacobian and then separately compute the JVP using the
+        tangent.
+        """
+
+        def wrapper(params):
+            new_tapes = [_copy_tape(t, a) for t, a in zip(tapes, params)]
+
+            with qml.tape.Unwrap(*new_tapes):
+                all_jacs = []
+                for new_t in new_tapes:
+                    jvp_tapes, fn = gradient_fn(new_t, shots=device.shots, **gradient_kwargs)
+                    jacs = execute_fn(jvp_tapes)[0]
+                    jacs = fn(jacs)
+                    all_jacs.append(jacs)
+
+            if len(all_jacs) == 1:
+                return all_jacs[0]
+
+            return all_jacs
+
+        expected_shapes = _jac_shape_dtype_tuple(tapes, device)
+        res = jax.pure_callback(wrapper, expected_shapes, params)
+        return res
```

Comment:
It looks good, maybe we could call it `jacobian_eval_transform_with_callback`which would be more descriptive.

Reply:
Just personal preference I guess, it looks good.